PSReadLine/ReadLine.cs

```diff
@@ -292,10 +278,12 @@ internal static PSKeyInfo ReadKey()
                 throw new OperationCanceledException();
             }
 
-            if (handleId == CancellationRequested)
+            if (_singleton._cancelReadCancellationToken.IsCancellationRequested)
             {
-                // ReadLine was cancelled. Save the current line to be restored next time ReadLine
-                // is called, clear the buffer and throw an exception so we can return an empty string.
+                // ReadLine was cancelled. Dequeue the dummy input sent by the host, save the current
+                // line to be restored next time ReadLine is called, clear the buffer and throw an
+                // exception so we can return an empty string.
+                _singleton._queuedKeys.Dequeue();
```

Comment:
Please update the comment at the line `handleId = WaitHandle.WaitAny(_singleton._requestKeyWaitHandles, 300)` -- remove the "cancellation" related line.